fastly.go

```diff
@@ -14,41 +13,47 @@ import (
 
 func getServiceNames(token string, serviceIDs []string, logger log.Logger) map[string]string {
 	serviceNames := map[string]string{}
-	for _, serviceID := range serviceIDs {
-		serviceNames[serviceID] = getServiceName(token, serviceID, log.With(logger, "service_id", serviceID))
-	}
-	return serviceNames
-}
 
-func getServiceName(token, serviceID string, logger log.Logger) string {
-	req, err := http.NewRequest("GET", fmt.Sprintf("https://api.fastly.com/service/%s/details", url.QueryEscape(serviceID)), nil)
+	req, err := http.NewRequest("GET", "https://api.fastly.com/service", nil)
 	if err != nil {
 		level.Error(logger).Log("err", err)
-		return serviceID
+		return serviceNames
```

Comment:
Hmm, this needs more thinking. If -service is optional, and this lookup process fails for whatever reason, the exporter won't do useful work. My intuition is that the program should exit in this case. And I think it's principle-of-least-surprise if the program would exit if getServiceNames failed, regardless if any -service is specified or not. What do you think?

Reply:
This is actually an intermediary in series of patches.

Patch+1: serviceNames maintained in a separate go routine and shared with the rt polling go routines via map & mutex or sync.map.  This addresses service name changes over time.

Patch+2: rt polling go routines actually instantiated and destroyed by the serviceNames go routine.  This addresses services added or removed over time, and will also do something like sleep/retry if the /service call or results are not fit.  **this should address your concerns above.**

Reply:
the "getServiceNames fitness & retry or exit" commit should address this in the interim.